edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/ILogsUploader.cs

```diff
@@ -1,10 +1,13 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Agent.Core.Logs
 {
+    using System;
     using System.Threading.Tasks;
 
     public interface ILogsUploader
     {
         Task Upload(string uri, string module, byte[] payload, LogsContentEncoding logsContentEncoding, LogsContentType logsContentType);
```

Comment:
Since `Upload` only gets called for `LogsContentType.JSON` and `GetUploaderCallback` only for `LogsContentType.Text`, does it make sense to remove their `LogsContentType` parameter and rename them to `UploadJSON` and `GetTextUploaderCallback` ?

Reply:
Sure, I could do that. But isn't it better to keep it generic?

Reply:
No, if it's intentionally generic then that's fine.

Reply:
yes, that was the idea..